tests/e2e/run-local.sh

```diff
@@ -29,44 +30,58 @@ usage() {
                          the tests. Defaults to "kata-qemu".
 	-u: undo the installation and configuration before exiting. Useful for
 	    baremetal machine were it needs to do clean up for the next tests.
+	-t: enable default timeout for each operation (useful for CI)
 	EOF
 }
 
 parse_args() {
-	while getopts "hr:u" opt; do
+	while getopts "hr:ut" opt; do
 		case $opt in
 			h) usage && exit 0;;
 			r) runtimeclass="$OPTARG";;
 			u) undo="true";;
+			t) timeout="true";;
 			*) usage && exit 1;;
 		esac
 	done
 }
 
+run() {
+	duration=$1; shift
+	if [ "$timeout" == "true" ]; then
+		timeout $duration "$@"
+	else
+		"$@"
+	fi
+}
+
 undo_changes() {
 	pushd "$script_dir" >/dev/null
 	# Do not try to undo steps that did not execute.
 	if [ $step_install_operator -eq 1 ]; then
-		echo "INFO: Uninstall the operator"
-		sudo -E PATH="$PATH" bash -c './operator.sh uninstall' || true
+		echo "::info:: Uninstall the operator"
+		run 10m sudo -E PATH="$PATH" bash -c './operator.sh uninstall' || true
 	fi
 
 	if [ $step_start_cluster -eq 1 ]; then
-		echo "INFO: Shutdown the cluster"
-		sudo -E PATH="$PATH" bash -c './cluster/down.sh' || true
+		echo "::info:: Shutdown the cluster"
+		run 5m sudo -E PATH="$PATH" bash -c './cluster/down.sh' || true
 	fi
 
 	if [ $step_bootstrap_env -eq 1 ]; then
-		echo "INFO: Undo the bootstrap"
-		ansible-playbook -i localhost, -c local --tags undo ansible/main.yml || true
+		echo "::info:: Undo the bootstrap"
+		run 5m ansible-playbook -i localhost, -c local --tags undo ansible/main.yml || true
 	fi
 	popd >/dev/null
 }
 
 on_exit() {
+	RET="$?"
 	if [ "$undo" == "true" ]; then
+		[ "$RET" -ne 0 ] && echo && echo "::error:: Testing failed with $RET, starting the cleanup..."
 		undo_changes
 	fi
+	[ "$RET" -ne 0 ] && echo && echo "::error:: Testing failed with $RET" || echo "::info:: Testing passed"
```

Comment:
When `undo` is true and `RET` is nonzero, we'll get a couple "testing failed" error messages (one before undoing changes, and one after). I guess that's OK, though this could be tweaked if you want.

Reply:
That was my intention, you get notified things went wrong (as it might not be obvious from the logs) and that things are going to be cleared, then you get a bunch of output spilled over finishing with ansible message that everything went well after which you get this second Testing failed message to emphasize although Ansible is happy the testing actually did not went well.